runtime/sema/type.go

```diff
@@ -2533,6 +2569,16 @@ func ArraySliceFunctionType(elementType Type) *FunctionType {
 	)
 }
 
+func ArrayToVariableSizedFunctionType(elementType Type) *FunctionType {
+	return NewSimpleFunctionType(
+		FunctionPurityImpure,
```

Comment:
I think it should be possible to make this `view`?

Reply:
cc @dsainati1

Reply:
Yeah, it should be

Reply:
Done, updated.